plugins/modules/win_dns_delegation.ps1

```diff
@@ -0,0 +1,113 @@
+#!powershell
+# Copyright: (c) 2021 Sebastian Gruber (@sgruber94) ,dacoso GmbH All Rights Reserved.
+# SPDX-License-Identifier: GPL-3.0-only
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#AnsibleRequires -CSharpUtil Ansible.Basic
+
+$spec = @{
+    options             = @{
+        name       = @{ type = "str"; required = $true }
+        zone       = @{ type = "str"; required = $true }
+        state      = @{ type = "str"; choices = "absent", "present"; default = "present" }
+        NameServer = @{ type = "str" }
+        IPAddress  = @{ type = "list"; elements = "str" }
+    }
+    supports_check_mode = $true
+}
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+$check_mode = $module.CheckMode
+$name = $module.Params.name
+$zone = $module.Params.zone
+$state = $module.Params.state
+$NameServer = $module.Params.NameServer
+$IPAddress = $module.Params.IPAddress
+
+$parms = @{
+    Name          = $zone
+    ChildZoneName = $name
+}
+Try {
+    # Import DNS Server PS Module
+    Import-Module DNSServer
+} Catch {
+    # Couldn't load the DNS Module
+    $module.FailJson("The DNSServer module failed to load properly: $($_.Exception.Message)", $_)
+}
+
+try {
+    $delegationzone = Get-DnsServerZoneDelegation @parms
+} catch {
+    $module.FailJson("Failed to get dns delegation zone. $($_.Exception.Message)", $_)
+}
+
+if ($state -eq "present") {
+    $parms.NameServer = $NameServer
+    $parms.IPAddress = $IPAddress
+    if ($check_mode) { $parms.WhatIf = $check_mode }
+    #check if exist
+    if ($delegationzone) {
+        if ($delegationzone[0].NameServer.RecordData.NameServer.trim(".") -notmatch $nameserver) {
+            try {
+                Add-DnsServerZoneDelegation @parms -PassThru -Confirm:$false
+                $module.Result.changed = $true
+            } catch {
+                $module.FailJson("Failed to add dns delegation zone $($name): $($_.Exception.Message)", $_)
```

Comment:
Same here, the try catch isn't really doing much here, just best off leaving it out.